### ISSUE TITLE
 /usr/local/etc/haproxy/haproxy.cfg Not Found when using Docker SWARM

### ISSUE THREAD

**Issue by AndrewMoneronGenex** [docker-library/haproxy#30]
Get a /usr/local/etc/haproxy/haproxy.cfg Not Found when creating a Service with Swarm

**Comment by yosifkit** [comment]
Did you build an image `FROM haproxy` that includes your config file? The default image does not contain a config file since it will be different for every user.

**Comment by AndrewMoneronGenex** [comment]
Affirm,  I managed to solve it by changing the location of the configuration file to / and changing the -f. But the default destination path wouldn't work.

**Comment by tianon** [comment]
I can't seem to replicate the failure with the default path that you're describing:
```Dockerfile
FROM haproxy

COPY haproxy.cfg /usr/local/etc/haproxy/
```
Starts up without issue:
```console
<7>haproxy-systemd-wrapper: executing /usr/local/sbin/haproxy -p /run/haproxy.pid -f /usr/local/etc/haproxy/haproxy.cfg -Ds 
```
(no further log since `haproxy` is running successfully)

**Comment by tianon** [comment]
In fact, our official `haproxy-basics` test uses that very location as well: https://github.com/docker-library/official-images/blob/c0fe268a6ec75c0bebbb225db7dc19964b469ae9/test/tests/haproxy-basics/run.sh#L17
```Dockerfile
COPY dir/haproxy.cfg /usr/local/etc/haproxy/
```

**Comment by AndrewMoneronGenex** [comment]
I think I figured it out.

When I first tested, I just used docker run with -v, pulling directly from your image.
The Second Time, I created a DockerFile and named it haproxy. Got the error.
I then took your whole docker file and entrypoint.sh, modified it to copy in the config file to the root dir and built with the name haproxy.
But still got cannot find /usr/local/etc/haproxy/haproxy.cfg. Thought this cant be right and dockers cache is getting confused with the image name.
Then Built as myproxy2 and all worked well still with the root dir but working.
